airflow_dbt/hooks/base.py

```diff
@@ -0,0 +1,164 @@
+    :param full_refresh: If specified, dbt will drop incremental models and
+        fully-recalculate the incremental table from the model definition
+    :type full_refresh: bool
+    :param data: Run data tests defined in "tests" directory.
+    :type data: bool
+    :param schema: Run constraint validations from schema.yml files
+    :type schema: bool
+    :param models: Flag used to choose a node or subset of nodes to apply
+        the command to (v0.210.0 and lower)
+    :type models: str
+    :param exclude: Nodes to exclude from the set defined with
+        select/models
+    :type exclude: str
+    :param select: Flag used to choose a node or subset of nodes to apply
+        the command to (v0.21.0 and higher)
+    :type select: str
+    :param selector: Config param to reference complex selects defined in
+        the config yaml
+    :type selector: str
+    :param output: {json,name,path,selector}
+    :type output: str
+    :param output_keys: Which keys to output
+    :type output_keys: str
+    :param host: Specify the host to listen on for the rpc server
+    :type host: str
+    :param port: Specify the port number for the rpc server
+    :type port: int
+    :param fail_fast: Stop execution upon a first test failure
+    :type fail_fast: bool
+    :param args:
+    :type args:
+    :param no_compile: Do not run "dbt compile" as part of docs generation
+    :type no_compile: bool
+    """
+    dbt_command_config_annotations = DbtCommandConfig.__annotations__
+    if not dbt_bin or not command:
+        raise ValueError("dbt_bin and command are mandatory")
+    command_params = []
+    for key, value in params.items():
+        # check that the key belongs to DbtCommandConfig keys
+        if key not in dbt_command_config_annotations.keys():
+            raise ValueError(f"{key} is not a valid key")
+        if value is not None:
+            # check that the value has the correct type from dbt_command_config_annotations
+            if type(value) != dbt_command_config_annotations[key]:
+                raise TypeError(f"{key} has to be of type {dbt_command_config_annotations[key]}")
+            # if the param is not bool it must have a non null value
+            cli_param_from_kwarg = "--" + key.replace("_", "-")
+            command_params.append(cli_param_from_kwarg)
+            if type(value) is str:
+                command_params.append(value)
+            elif type(value) is int:
+                command_params.append(str(value))
+            elif type(value) is dict:
+                command_params.append(json.dumps(value))
+            elif type(value) is bool:
+                if not value:
+                    raise ValueError(
+                        f"`{key}` cannot be false. Flags will be passed always "
+                        f"afirmatively. If you want to use a negative flag "
+                        f"such as --no-use-colors then provide "
+                        f"`no_use_colors=True`")
```

Comment:
I see...

I do think the prior one is more readable. It even saves LOCs!

When you get this you have to "compile" it on your head for every single case. There's a finite amount of cases since there's not many types in python :P (just done some things in golang were you have `int`, `int16`, `int32`, `int64`... and same for floats, uints, ufloats... crazy!)

Arguably what your brain will compile this to is the same code it comes from.

```python
    def raise_error_if_false(key: str, value: bool) -> None:
        if not value:
            raise ValueError(
                f"`{key}` cannot be false. Flags will be passed always "
                "afirmatively. If you want to use a negative flag such as "
                "--no-use-colors then provide `no_use_colors=True`"
            )

    parsing_mapper = {
        str: lambda _, v: v,
        int: lambda _, v: str(v),
        dict: lambda _, v: json.dumps(v),
        bool: lambda k, v: raise_error_if_false(k, v)
    }

    mapper = parsing_mapper.get(type(value))
    mapped_value = mapper(key, value)
    command_params.append(mapped_value)
```

there's only 4 cases. But if those cases where to grow I will for sure do it with "tabular" style